sigstore/_internal/oidc.py

```diff
@@ -54,3 +66,173 @@ def __init__(self, identity_token: str) -> None:
+        r = urllib.parse.urlsplit(self.path)
+
+        # Handle auth response
+        if r.path == server.redirect_path:
+            self.send_response(200)
+            self.send_header("Content-Type", "text/html; charset=utf-8")
+            body = AUTH_SUCCESS_HTML.encode("utf-8")
+            self.send_header("Content-Length", str(len(body)))
+            self.end_headers()
+            self.wfile.write(body)
+            server.auth_response = urllib.parse.parse_qs(r.query)
+            return
+
+        # Any other request generates an auth request
+        url = server.auth_request()
+        self.send_response(302)
+        self.send_header("Location", url)
+        self.end_headers()
+
+
+OOB_REDIRECT_URI = "urn:ietf:wg:oauth:2.0:oob"
+
+
+class RedirectServer(http.server.HTTPServer):
+    def __init__(self):
+        super().__init__(("127.0.0.1", 0), RedirectHandler)
+        self.state = None
+        self.nonce = None
+        self.auth_response = None
+        self._port: int = self.socket.getsockname()[1]
+        self.is_oob = False
+
+    @property
+    def active(self) -> bool:
+        return self.auth_response is None
+
+    @property
+    def base_uri(self) -> str:
+        return f"http://localhost:{self._port}"
+
+    @property
+    def redirect_path(self) -> str:
+        return "/auth/callback"
+
+    @property
+    def redirect_uri(self) -> str:
+        return (
+            (self.base_uri + self.redirect_path)
+            if not self.is_oob
+            else OOB_REDIRECT_URI
+        )
+
+    def generate_code_challenge(self) -> bytes:
+        self.code_verifier = base64.urlsafe_b64encode(os.urandom(32)).rstrip(b"=")
+        return base64.urlsafe_b64encode(
+            hashlib.sha256(self.code_verifier).digest()
+        ).rstrip(b"=")
+
+    def auth_request_params(self) -> Dict[str, str]:
+        code_challenge = self.generate_code_challenge()
+        self.state = str(uuid.uuid4())
+        self.nonce = str(uuid.uuid4())
+        return {
+            "response_type": "code",
+            "client_id": "sigstore",
+            "scope": "openid email",
+            "redirect_uri": self.redirect_uri,
+            "code_challenge": code_challenge.decode("utf-8"),
+            "code_challenge_method": "S256",
+            "state": self.state,
+            "nonce": self.nonce,
+        }
+
+    def auth_request(self) -> str:
+        params = self.auth_request_params()
+        return "https://oauth2.sigstore.dev/auth/auth?" + urllib.parse.urlencode(params)
+
+    def enable_oob(self) -> None:
+        self.is_oob = True
+
+
+def get_identity_token() -> str:
+    """
+    Retrieve an OpenID Connect token from the Sigstore provider
+
+    This function and the components that it relies on are based off of:
+    https://github.com/psteniusubi/python-sample
+    """
+
+    code: str
+    with RedirectServer() as server:
+        # Launch web browser
+        if not webbrowser.open(server.base_uri):
```

Comment:
```suggestion
        if webbrowser.open(server.base_uri):
```

Reply:
[cecc6c6](https://github.com/sigstore/sigstore-python/pull/56/commits/cecc6c6c7b330812f9a5fb904704ce540fc898ec)

Reply:
Oops. Did that to test the out-of-band case but then left it that way. 🤦